homeassistant/components/climate/zha.py

```diff
@@ -0,0 +1,251 @@
+"""
+Climate on Zigbee Home Automation networks.
+
+For more details on this platform, please refer to the documentation
+at https://home-assistant.io/components/climate.zha/
+"""
+
+import asyncio
+import logging
+import time
+
+from homeassistant.components import zha
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate import (
+    DOMAIN, STATE_ECO, ATTR_OPERATION_MODE)
+from homeassistant.const import (
+    STATE_OFF, TEMP_CELSIUS, ATTR_TEMPERATURE)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['zha']
+ATTR_HEATING_POWER = 'heating_power'
+STATE_COMFORT = 'comfort'
+CLEAN_TIME_ZONE = 1
+OPERATIONS = {0: STATE_OFF,
+              4: STATE_COMFORT,
+              5: STATE_ECO}
+REVERSE_OPERATIONS = dict((v, k) for k, v in OPERATIONS.items())
+OPERATION_LIST = list(OPERATIONS.values())
+OFF_CODE = 0x00
+COMFORT_CODE = 0x04
+ECO_CODE = 0x05
+NO_ERROR_ANSWER_CODE = 0x00
+
+
+@asyncio.coroutine
+def safe_read(cluster, attributes, raw=False):
+    """Swallow all exceptions from network read.
+
+    If we throw during initialization, setup fails. Rather have an
+    entity that exists, but is in a maybe wrong state, than no entity.
+
+    .. note:: Could be move to homeassistant/components/zha/__init__.py
+    """
+    try:
+        result, _ = yield from cluster.read_attributes(
+            attributes,
+            allow_cache=False,
+            raw=raw,
+        )
+        return result
+    except Exception:  # pylint: disable=broad-except
```

Comment:
I think this should at least print an error message if a read fails.

Reply:
Couldn't we also pinpoint the exception to eg `OSError`?